
Create a program that gets from user 4 inputs that represent the expenses ("Transportation", "Eating out", "Groceries", "Health and cosmetics"). The program should calculate the total expense.



In [2]:
Transportation=int(input('Daxil edin: '))
Eating=int(input('Daxil edin: '))
Groceries=int(input('Daxil edin: '))
Health=int(input('Daxil edin: '))
total=0
while True:
    total=Transportation+Eating+Groceries+Health
    print(total)
    break

Daxil edin: 23
Daxil edin: 34
Daxil edin: 6
Daxil edin: 43
106


Create a program where user enters a password. If the password is "bigsecret" display "Access granted", otherwise display "Access denied".

In [7]:
password=input('Daxil edin: ')
while True:
    if password=="bigsecret":
        print('Access granted')
        break
    else:
        print('Access denied')
        break

Daxil edin: bigsecret
Access granted


Create a program that asks user a question until the user answers it correctly.

In [16]:
while True:
    answer=input('Saat necedir?: ')
    if answer=='Saat 5 dir':
        print('Dogrudur')
        break

Saat necedir?: 3 dur
Saat necedir?: 8 dir
Saat necedir?: Saat 5 dir
Dogrudur


Create a program that takes two inputs from a user: username and password. The program should ask for these inputs until each field is valid. The username must be at least 6 characters long. Password must be at least 8 characters long and contain a word "anonim" or "anon" in it.

In [19]:
while True:
    password=input('Daxil edin: ')
    username=input('Daxil edin: ')
    if len(password)>=8 and len(username)>=6 and ('anonim'==username or 'anon'==username):
        print('Qebul edildiniz')
        break
    else:
        print('Yeniden cehd edin')
        

Daxil edin: esmerler
Daxil edin: anonim
Qebul edildiniz


Write a program that displays how many thousands, hundreds, tens, ones (units) are in the number given by user (Assume that user always enters integers).

Example:

Number: 1437

Thousands: 1

Hundreds: 4

Tens: 3

Ones: 7

**Bu Task While-a aid deyil**

In [2]:
number=int(input('Daxil edin: '))

Thousands=[]
Hundreds=[]
Tens=[]
Ones=[]

a=number//1000
Thousands.append(a)

b=number//100
Hundreds.append(b)

c=number//10
Tens.append(c)

d=number//1
Ones.append(d)

print(f" Thousands= {Thousands} , Hundreds= {Hundreds} , Tens= {Tens} , Ones= {Ones}" )

Daxil edin: 1234
 Thousands= [1] , Hundreds= [12] , Tens= [123] , Ones= [1234]


Write a program that displays all odd numbers between 0 and 100

VERSION 2

The end point is determined by the user

VERSION 3

Display the amount of odd numbers as well

In [7]:
# elave qeyd. user dediyi edede qeder tek ededleri gostersin

user=int(input('Daxil edin: '))
i=-1
while i<user:
    i+=2
    print(i)

Daxil edin: 23
1
3
5
7
9
11
13
15
17
19
21
23


In [16]:
# user dediyi ededden geriye 3-3 azalsin amma menfi qiymetler almasin.

user = int(input("Daxil edin: "))

while user > 0:
    user -= 3
    if user >= 0:
        print(user)

Daxil edin: 23
20
17
14
11
8
5
2


In [ ]:
#version 1 --- Write a program that displays all odd numbers between 0 and 100
i=0
while i<100:
    i+=1
    print(i)
    

In [51]:
# version 2  --- the end point is determined by the user
user=int(input('daxil edin: '))
i=0
while i<user:
    i+=1
    print(i)
    


daxil edin: 23
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23


In [23]:
#version 3 --- Display the amount of odd numbers as well
user = int(input("Daxil edin: "))
i=1
cem=0
while i<=user:
    if i%2==1:
        cem+=1
    i+=1
print(f"Tek ededlerin sayi: {cem}")


Daxil edin: 23
Tek ededlerin sayi: 12


Write a program that adds all numbers until 20 inclusive

1 + 2 + 3 + 4 ... + 20 = 210

In [45]:
i=0
cem=0
while i<23:
    i+=1
    cem+=i
print(cem)

276


Write a program that takes three numbers as an input (a, b, c) and checks how many numbers between a and b are divisible by c.

Example:

a: 0

b: 10

c: 2

There are 5 numbers divisible by 2 between 0 and 10

In [10]:
a=int(input('daxil edin: '))
b=int(input('daxil edin: '))
c=int(input('daxil edin: '))
say=0
for i in range(a,b):
    if i%c==0:
        say+=1
print(say)
            

daxil edin: 0
daxil edin: 10
daxil edin: 2
5


Create a program that asks user to enter a password. The password must me at least 6 characters long. The program does not stop until user enters a valid password

Example:

Password: 123

Invalid password, try again... Password: 123456

Success

In [14]:
while True:
    user= input ('Daxil edin: ')
    if len(str(user))>=6:
        print('Dogrudur')
        break
    else: 
        print('Yeniden cehd edin')

Daxil edin: 123
Yeniden cehd edin
Daxil edin: 123456
Dogrudur


Write a program that tells whether a triangle with given sides exists or not.


Example

Side a: 5

Side b: 6

Side c: 7

Triangle with sides 5, 6, 7 exists

Side a: 10

Side b: 4

Side c: 3

Triangle with sides 10, 4, 3 DOES NOT exist

In [15]:
while True:
    a = int(input('Daxil edin a: '))
    b = int(input('Daxil edin b: '))
    c = int(input('Daxil edin c: '))
    
    if (a+b>c) and (b+c>a) and (a+c>b):
        print("Belə üçbucaq mövcuddur.")
        break
    else:
        print("Belə üçbucaq mövcud deyil.")

Daxil edin a: 3
Daxil edin b: 4
Daxil edin c: 5
Belə üçbucaq mövcuddur.


Create a program that asks the user to guess a randomly generated secret number (from 1 to 12 inclusive). If the user enters a number that is less or greater than secret number, display corresponding message. The user can guess only 4 times. The program doesn't stop until the user guesses the secret number or the attempt limit is reached. Display the amount of attempts it took for the user to guess the secret number.

Example:

Your guess: 5

Go higher

Your guess: 12

Go lower

Your guess: 10

You won!

It took you 3 attempts to win the game

Another example:

Your guess: 2

Go higher

Your guess: 3

Go higher

Your guess: 12

Go lower

Your guess: 11

You lost... the secret number is 10

In [31]:
limit=4
cehd=0
while cehd<limit:
    a = int(input('Sizin təxmininiz: '))
    b=23
    if a<b:
        print('Yuxari qalxin')
    if a>b:
        print('Asagi enin')
    elif a==b:
        print('Siz dogru tapdiniz!')
    cehd+=1
        
        
if cehd==limit:
    print('Siz uduzdunuz')

Sizin təxmininiz: 45
Asagi enin
Sizin təxmininiz: 67
Asagi enin
Sizin təxmininiz: 56
Asagi enin
Sizin təxmininiz: 78
Asagi enin
Siz uduzdunuz


In [33]:
limit=4
cehd=0
while cehd<limit:
    a = int(input('Sizin təxmininiz: '))
    b=23
    if a<b:
        print('Yuxari qalxin')
    if a>b:
        print('Asagi enin')
    elif a==b:
        print('Siz dogru tapdiniz!')
    cehd+=1
        
        
if cehd==limit:
    print('Siz uduzdunuz')

Sizin təxmininiz: 45
Asagi enin
Sizin təxmininiz: 56
Asagi enin
Sizin təxmininiz: 23
Siz dogru tapdiniz!


KeyboardInterrupt: Interrupted by user